src/app/containers/FrontPage/index.test.jsx

```diff
@@ -59,5 +62,57 @@ describe('FrontPageContainer', () => {
         <FrontPageContainer />,
       );
     });
+    describe('Assertions', () => {
+      let FrontPageComponent;
+      beforeAll(() => {
+        jest.resetModules();
+        jest.unmock('../PageHandlers/withError');
+        jest.unmock('../PageHandlers/withLoading');
+        jest.unmock('../PageHandlers/withData');
+
+        jest.mock('react', () => {
+          const original = jest.requireActual('react');
+          return {
+            ...original,
+            useContext: jest.fn(),
+          };
+        });
+
+        const { useContext } = jest.requireMock('react');
+        useContext.mockReturnValue(igboConfig);
+        FrontPageComponent = jest.requireActual('.').default;
+      });
+
+      it('should not render frontpage if still loading', () => {
+        const { container } = render(<FrontPageComponent loading />);
+        const { textContent } = container.querySelector('main');
+
+        expect(textContent).toEqual('');
+        expect(textContent).not.toContain('FrontPageMain');
+      });
+
+      it('should render error page when an error occurs', () => {
+        const { container } = render(
+          <FrontPageComponent error="An error Occured" />,
+        );
+
+        const { textContent } = container.querySelector('main');
+        expect(textContent).toContain('Internal server error');
+        expect(textContent).not.toContain('FrontPageMain');
+      });
+
+      it('should render the frontpage with data', () => {
+        const data = {
+          pageData: igboData,
+          status: 200,
+        };
+
+        const { container } = render(
+          <FrontPageComponent error="" data={data} service="igbo" />,
+        );
+
+        expect(container.textContent).toEqual('FrontPageMain');
```

Comment:
this test will pass even if data breaks upstream and isnt passed down to FrontPageComponent, could it maybe have a `toHaveBeenCalledWith` assertion please?